src/S3Bucket/Modals/RenameModal.vue

```diff
@@ -0,0 +1,147 @@
+<template>
+  <VDialog
+    v-model="renameDialog"
+    width="1024"
+    @update:model-value="(val) => !val && onCancel()"
+  >
+    <template #activator="{ props: renameProps }">
+      <VBtn v-bind="renameProps" icon="mdi-pencil-circle" title="Rename" />
+    </template>
+    <VCard class="pa-3">
+      <VForm
+        @submit.prevent="renameObject"
+        @update:model-value="(val) => (formIsValid = val)"
+      >
+        <VCardTitle class="w-100 d-inline-flex justify-space-between text-h5">
+          Rename Object
+          <VBtn
+            icon="mdi-close"
+            title="Close this dialog"
+            variant="text"
+            @click="onCancel"
+          />
+        </VCardTitle>
+        <VCardText>
+          <VTextField
+            v-model="renameNew"
+            label="New Object Name"
+            placeholder="Enter name here"
+            type="text"
+            :rules="requiredRule"
+            required
+          />
+        </VCardText>
+        <VCardActions class="d-flex justify-end px-3 mb-1">
+          <VTooltip location="start" :text="formError">
+            <template #activator="{ props: activatorProps }">
+              <VIcon
+                v-if="formError"
+                v-bind="activatorProps"
+                color="error"
+                size="large"
+                icon="mdi-alert-circle"
+                class="mt-1"
+              />
+            </template>
+          </VTooltip>
+          <VBtn
+            prepend-icon="mdi-close"
+            variant="flat"
+            size="large"
+            class="px-4 mx-2"
+            @click="onCancel"
+            >Cancel</VBtn
+          >
+          <VBtn
+            :loading="isSubmitting"
+            :disabled="!formIsValid"
+            type="submit"
+            variant="flat"
+            color="primary"
+            size="large"
+            :width="isSubmitting ? '150' : '100'"
+            class="px-4"
+            >Rename
+            <template #loader>Submitting…</template>
+          </VBtn>
+        </VCardActions>
+      </VForm>
+    </VCard>
+  </VDialog>
+</template>
+
+<script setup>
```

Comment:
There's so much similarity between all these modals that I wonder what you could do to DRY them out. They all seem to use some of the same ref ideas (isLoading, isDialogOpen, errorMessage) and somewhat similar fetch functions that could maybe move to a composable. Or maybe they also have enough shared template that a shared parent component with a content slot would be better.

Thoughts? It may not be worth it or they may be different enough to warrant some duplication, but I wanted to ask.

Reply:
I think there's enough differences happening that we're not going to get the most out of making a separate component or composable